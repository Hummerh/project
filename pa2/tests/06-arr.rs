enum Lst {
    Empty,
    Cons(i32, Box<Lst>)
}

fn main() {
    let a = [1, 2, 3];

    let [mut x, y, z] = a;

    /*
     * println!("arr: {} {} {}\n", a[0], a[1], a[2]);
     */

    let list = Lst::Cons(1,
        Box::new(Lst::Cons(2,
            Box::new(Lst::Cons(3,
                Box::new(Lst::Empty))))));


    match (list) {
        Lst::Empty => { b"what"; },
        Lst::Cons(_, l) => { b"nope"; }
    }
}
