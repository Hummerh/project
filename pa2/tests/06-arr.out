(crate
    (items
        (enum-def
            (id
                (Lst))
            (enum-ctor-defs
                (enum-ctor-def
                    (id
                        (Empty)))
                (enum-ctor-def
                    (id
                        (Cons))
                    (enum-ctor-params
                        (type-i32)
                        (type-box
                            (id
                                (Lst)))))))
        (fn-def
            (id
                (main))
            (block
                (let
                    (pat-id
                        (id
                            (a)))
                    (arr
                        (exprs
                            (lit-dec)
                            (lit-dec)
                            (lit-dec))))
                (let
                    (pat-arr
                        (pat-arr-elems
                            (pat-mut-id
                                (id
                                    (x)))
                            (pat-id
                                (id
                                    (y)))
                            (pat-id
                                (id
                                    (z)))))
                    (id
                        (a)))
                (let
                    (pat-id
                        (id
                            (list)))
                    (enum
                        (enum-ctor
                            (id
                                (Lst))
                            (id
                                (Cons)))
                        (exprs
                            (lit-dec)
                            (box-new
                                (exprs
                                    (enum
                                        (enum-ctor
                                            (id
                                                (Lst))
                                            (id
                                                (Cons)))
                                        (exprs
                                            (lit-dec)
                                            (box-new
                                                (exprs
                                                    (enum
                                                        (enum-ctor
                                                            (id
                                                                (Lst))
                                                            (id
                                                                (Cons)))
                                                        (exprs
                                                            (lit-dec)
                                                            (box-new
                                                                (exprs
                                                                    (enum
                                                                        (enum-ctor
                                                                            (id
                                                                                (Lst))
                                                                            (id
                                                                                (Empty)))))))))))))))))
                (match
                    (id
                        (list))
                    (match-arms
                        (match-arm
                            (pats
                                (pat-enum
                                    (enum-ctor
                                        (id
                                            (Lst))
                                        (id
                                            (Empty)))))
                            (block
                                (lit-str)))
                        (match-arm
                            (pats
                                (pat-enum
                                    (enum-ctor
                                        (id
                                            (Lst))
                                        (id
                                            (Cons)))
                                    (pat-enum-ctor-params
                                        (pat-wild)
                                        (pat-id
                                            (id
                                                (l))))))
                            (block
                                (lit-str)))))))))